src/util.ts

```diff
@@ -198,3 +198,26 @@ export const renameKeys = (doc: Document, keys: Record<string, string>) => {
     }
   }
 }
+
+const castError = (err: unknown): Error => {
+  if (err instanceof Error) {
+    return err
+  }
+  if (typeof err === 'string') {
+    return new Error(err)
+  }
+  return new Error('Unknown error')
```

Comment:
Might be good to use the error option for cause as described [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause#providing_structured_data_as_the_error_cause) so that we don't lose the original thrown value in this case

Reply:
I see that we use [make-error](https://github.com/JsCommunity/make-error) in some places. It would be good to test the behavior with those kinds of errors, and make sure that this doesn't obscure the original error message.

Reply:
Oh, nice 👍